dev/release/verify-yum.sh

```diff
@@ -220,7 +220,7 @@ pushd build/minimal_build
 ${cmake_command} .
 make -j$(nproc)
 ./arrow-example
-c++ -o arrow-example example.cc $(pkg-config --cflags --libs arrow) -std=c++17
+c++ -o arrow-example example.cc $(pkg-config --cflags --libs arrow) -std=c++20
```

Comment:
For old C++ compiler:

```suggestion
c++ -o arrow-example example.cc $(pkg-config --cflags --libs arrow) -std=c++2a
```